File: tests/Java/CaesarTest.java
import static org.junit.Assert.*;
import java.util.ArrayList;
import java.lang.Throwable;
import java.lang.Exception;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;


public class CaesarTest
{

    private Alphabet alphabet;
    @Before
    public void setUp()
    {
        try{
            alphabet = new Alphabet("../../../etc/Frequency/English.csv");
        }catch(Exception e)
        {
            e.printStackTrace();
        }
    } 
   
    @Test 
    public void testSanity()
    {
        try
        {
            Caesar a = new Caesar(3, alphabet);
            CaesarExploits c = new CaesarExploits(alphabet);
            assertEquals("encrypt sanity check", "khoor", a.encrypt("hello"));
            assertEquals("decrypt sanity check", "hello", a.decrypt("khoor"));
            //c.ciphertextOnly("khoor"); //need to fix this function to return an arraylist
            assertEquals("known plaintext sanity check", 3, c.knownPlaintext("hello", "khoor"));
            assertEquals("chosen plaintext sanity check", 3, c.chosenPlaintext(a));
            assertEquals("chosen ciphertext sanity check", 3,c.chosenCiphertext(a));
        }catch(Exception e)
        {
            e.printStackTrace();
        }
    }
       
     
} 

File: tests/Python/CaesarTest.py
from Caesar import *
from CaesarExploits import *
import unittest

class TestCaesar(unittest.TestCase):
    
    def setup(self):
        self.alphabet =  ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
   
    def test_sanity(self):
        c = Caesar(3)
        e = CaesarExploits()
        self.assertTrue(c.encrypt("hello")=="khoor") 
        self.assertTrue(c.decrypt("khoor")=="hello")
        self.assertTrue(e.known_plaintext("hello", "khoor") == 3)
        self.assertTrue(e.chosen_plaintext(c)==3)
        self.assertTrue(e.chosen_ciphertext(c)==3)

if __name__ == "__main__":
    unittest.main()
